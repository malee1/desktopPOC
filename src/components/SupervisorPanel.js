import { React, useState } from 'react';
import { Button, Drawer, Box, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    margin: '0px',
    padding: '20px',
    width: '120px',
    // height: '100px',
    // backgroundColor: 'lightgrey',
    backgroundColor: 'rgb(2,0,36)',
    background:
      'linear-gradient(180deg, rgba(2,0,36,1) 0%, rgba(189,189,201,1) 0%, rgba(98,150,214,1) 100%)',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
    // justifyContent: 'center'
  },
  button: {
    width: '80px',
    marginTop: '10px'
  },
  listItem: {
    backgroundColor: 'red'
  },
  paper: {
    width: 'inherit',
    display: 'flex',
    alignItems: 'center'
  },
  image: {
    width: 'inherit'
  }
});

export default function SupervisorPanel() {
  const classes = useStyles();

  const [openButtonOne, setOpenButtonOne] = useState(false);
  const [openButtonTwo, setOpenButtonTwo] = useState(false);

  const handleOpenButtonOne = () => {
    setOpenButtonOne(true);
  };

  const handleOpenButtonTwo = () => {
    setOpenButtonTwo(true);
  };

  const handleClose = () => {
    setOpenButtonOne(false);
    setOpenButtonTwo(false);
  };

  return (
    <div className={classes.root} data-testid="supervisor-panel">
      <Box>
        <Typography variant="h5">Supervisor</Typography>
      </Box>
      <Button
        className={classes.button}
        variant="contained"
        color="primary"
        onClick={handleOpenButtonOne}
        data-testid="button"
      >
        View Metrics
      </Button>
      <Drawer
        style={{ width: '30vw' }}
        variant="temporary"
        anchor="right"
        open={openButtonOne}
        classes={{ paper: classes.paper }}
        data-testid="metrics-drawer"
      >
        <Typography variant="h5" data-testid="metrics-text">
          Content 1
        </Typography>
        <img
          className={classes.image}
          src="/images/metrics.png"
          alt="placeholder metrics"
          data-testid="image"
        />
        <Typography data-testid="metrics-text">Metrics</Typography>
        <Typography data-testid="metrics-text">Metrics</Typography>
        <Typography data-testid="metrics-text">Metrics</Typography>
        <Typography data-testid="metrics-text">Metrics</Typography>
        <Typography data-testid="metrics-text">Metrics</Typography>
        <Button
          className={classes.button}
          onClick={handleClose}
          variant="contained"
          color="secondary"
          data-testid="close-button"
        >
          Close
        </Button>
      </Drawer>

      <Button
        className={classes.button}
        variant="contained"
        color="primary"
        onClick={handleOpenButtonTwo}
        data-testid="button"
      >
        View Links
      </Button>
      <Drawer
        style={{ width: '30vw' }}
        variant="temporary"
        anchor="right"
        open={openButtonTwo}
        classes={{ paper: classes.paper }}
        data-testid="links-drawer"
      >
        <Typography variant="h5" data-testid="links-text">
          Content 2
        </Typography>
        <br />
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Typography data-testid="links-text">Useful Links</Typography>
        <Button
          className={classes.button}
          onClick={handleClose}
          variant="contained"
          color="secondary"
          data-testid="close-button"
        >
          Close
        </Button>
      </Drawer>
    </div>
  );
}
