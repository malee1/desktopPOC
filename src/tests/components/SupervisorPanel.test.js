import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

import SupervisorPanel from '../../components/SupervisorPanel';

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('Panel renders correctly', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<SupervisorPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders one component', () => {
    const { getAllByTestId } = render(<SupervisorPanel />);
    const panel = getAllByTestId('supervisor-panel');
    expect(panel.length).toBe(1);
  });
});

describe('it renders 2 buttons', () => {
  it('has two buttons', () => {
    const { getAllByTestId } = render(<SupervisorPanel />);
    const buttons = getAllByTestId('button');
    expect(buttons.length).toBe(2);
  });
});

describe('the metrics panel displays correctly', () => {
  it('shows the metrics drawer when the metrics b utton  is clicked', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Metrics');
    fireEvent.click(btn);
    const drawer = getAllByTestId('metrics-drawer');
    expect(drawer.length).toBe(1);
  });
  it('has the right text in the metrics panel', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Metrics');
    fireEvent.click(btn);
    const text = getAllByTestId('metrics-text');
    expect(text.length).toBe(6);
  });
  it('displays the placeholder image', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Metrics');
    fireEvent.click(btn);
    const img = getAllByTestId('image');
    expect(img.length).toBe(1);
  });
  it('closes the drawer when close button clicked', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Metrics');
    fireEvent.click(btn);
    const closeBtn = getByText('Close');
    fireEvent.click(closeBtn);
    const buttons = getAllByTestId('button');
    expect(buttons.length).toBe(2);
  });
});

describe('the links panel displays correctly', () => {
  it('shows the links drawer when the links button is clicked', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Links');
    fireEvent.click(btn);
    const drawer = getAllByTestId('links-drawer');
    expect(drawer.length).toBe(1);
  });
  it('has the right text in the links panel', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Links');
    fireEvent.click(btn);
    const text = getAllByTestId('links-text');
    expect(text.length).toBe(9);
  });
  it('closes the drawer when close button clicked', () => {
    const { getByText, getAllByTestId } = render(<SupervisorPanel />);
    const btn = getByText('View Links');
    fireEvent.click(btn);
    const closeBtn = getByText('Close');
    fireEvent.click(closeBtn);
    const buttons = getAllByTestId('button');
    expect(buttons.length).toBe(2);
  });
});
