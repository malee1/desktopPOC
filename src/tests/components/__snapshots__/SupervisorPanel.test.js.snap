// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="supervisor-panel"
>
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Supervisor
    </h5>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      View Metrics
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      View Links
    </span>
  </button>
</div>
`;
